app/Filters/ThreadFilters.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Filters;
+
+use App\Models\User;
+use Illuminate\Http\Request;
+
+class ThreadFilters
+{
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var
+     */
+    protected $builder;
+
+    /**
+     * @var string[]
+     */
+    protected $filters = ['by'];
+
+    /**
+     * ThreadFilters constructor.
+     */
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+    }
+
+    /**
+     * @param $builder
+     *
+     * @return mixed
+     */
+    public function apply($builder)
+    {
+        $this->builder = $builder;
+        foreach ($this->filters as $filter) {
+            if ($this->hasFilter($filter)) {
+                $this->$filter($this->request->$filter);
+            }
+        }
+
+        return $this->builder;
+    }
+
+    /**
+     * @param $username
```

Comment:
username is usually a string so should be `@param string $username` and `protected function by(string $username)`